src/main/java/org/folio/circulation/domain/validation/RequestLoanValidator.java

```diff
@@ -37,4 +51,31 @@ public RequestLoanValidator(LoanRepository loanRepository) {
         return singleValidationError(new ValidationError(message, parameters));
       }).map(loan -> requestAndRelatedRecords));
   }
+
+  public CompletableFuture<Result<RequestAndRelatedRecords>> refuseWhenUserHasAlreadyBeenLoanedAtLeastOneItem(
+    RequestAndRelatedRecords requestAndRelatedRecords) {
+
+    final Request request = requestAndRelatedRecords.getRequest();
+
+    CompletableFuture<Result<Collection<Item>>> itemsByInstanceId = itemByInstanceIdFinder.getItemsByInstanceId(UUID.fromString(request.getInstanceId()));
+    CompletableFuture<Result<MultipleRecords<Loan>>> openLoansByUserIdWithItem = loanRepository.findOpenLoansByUserIdWithItem(LOANS_PAGE_LIMIT, request.getUserId());
+
+    return itemsByInstanceId
+      .thenCombine(openLoansByUserIdWithItem, (collectionResult, multipleRecordsResult) -> {
+
+        List<String> itemIds = collectionResult.value().stream()
+          .map(Item::getItemId)
+          .collect(Collectors.toList());
+
+        return multipleRecordsResult.value().getRecords().stream()
+          .anyMatch(loan -> itemIds.contains(loan.getItemId()));
+    }).thenApply(isFound -> Result.succeeded(isFound)
+        .failWhen(found -> of(() -> found),
+          bool -> {
+            String message = "This requester has some item of instance on loan.";
+            ValidationError error = new ValidationError(message, "userId", request.getUserId());
+            return new ValidationErrorFailure(error);
+          })
```

Comment:
extract to a separate method please

Reply:
done

Reply:
I meant to move the only body, and leave the lambda argument in the method above

Reply:
bool -> createValidationErrorFailure(request))

private ValidationErrorFailure createValidationErrorFailure(Request request) {
    String message = "This requester has some item of instance on loan.";
    ValidationError error = new ValidationError(message, "userId", request.getUserId());
    return new ValidationErrorFailure(error);
  }
  
  right ?

Reply:
done